server/src/test/java/org/elasticsearch/common/settings/KeyStoreWrapperTests.java

```diff
@@ -104,6 +107,57 @@ public void testUpgradeNoop() throws Exception {
         assertEquals(seed.toString(), keystore.getString(KeyStoreWrapper.SEED_SETTING.getKey()).toString());
     }
 
+    public void testFailWhenStreamNotConsumed() throws Exception {
```

Comment:
This seems to test just one case. Can it be duplicated/refactored so it tests each of the readFully calls?